### PR TITLE
paginate passing tests. a few doc issues outstanding

### PR DESCRIPTION
**Pull request by gmbecker**

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (Melkiades) [inline .pre-commit-config.yaml:17]: I believe that I introduced it as it is very useful to check inputs etc... but I was thinking we should discuss it. It is used only in `toString` and imported for all of its functions, a bit too much probably. Still is imported by all other packages so it is not a game changer from the user's perspective imo

File: R/generics.R
Comment (Melkiades) [inline R/generics.R:535]:
```suggestion
#' @param obj The object to be paginated.
```

File: R/generics.R
Comment (Melkiades) [inline R/generics.R:543]:
```suggestion
#'
```

**Comment by shajoezhu** [comment]
Hi @gmbecker , thanks for the change, can you please add more test code. The auto-validate package has a threshold at 80%.

File: R/page_size.R
Comment (shajoezhu) [inline R/page_size.R:97]: @clarkliming fyi

File: NEWS.md
Comment (Melkiades) [inline NEWS.md:8]:
```suggestion
 * fix to test that failed on old Windows CRAN machine due to imperfect UTF support there.
```

File: NEWS.md
Comment (Melkiades) [inline NEWS.md:5]:
```suggestion
 * New `export_as_txt` function which can operate on any object with an applicable `matrix_form` method (including pagination).
 * new `paginate_indices` and `paginate_to_mpfs` functions.
```

File: NEWS.md
Comment (Melkiades) [inline NEWS.md:3]:
```suggestion
 * Margins are now interpreted as inches when specified (including the default).
```
again GitHub actions

File: NEWS.md
Comment (Melkiades) [inline NEWS.md:2]:
```suggestion
 * `font_size` is now 12 across the board, instead of being inconsistent between 8 and 12.
```
from GitHub Actions
/ grammar

NEWS.md#L2

"it was" is wordy or unneeded

Comment (Melkiades) [reply NEWS.md:2]: hopefully it works

**Review by Melkiades — CHANGES_REQUESTED** [review]
Thank you Gabe for the amazing job! We absolutely need to go through this PR together. So @ayogasekaram and I can understand better what are the fundamental changes. The news file was very helpful for me to clarify the changes. I am testing it now downstream, but we need to address lintr errors that may accumulate afterward. Blocking this until tests for rtables pass ;)

I believe this is an important refactor and we should have a version update. This makes it easier to test downstream too

**Review by Melkiades — APPROVED** [review]
Great Job @gmbecker! With current fixes in rtables, checks and tests pass also downstream for all packages (including chevron and citril), but it fails for scda.test (easy fix) and teal.widgets (no idea there)

Errors in teal.widgets:
```
* checking tests ...  Running ‘testthat.R’ ERROR
Running the tests in ‘tests/testthat.R’ failed.
Last 13 lines of output:
   29. │       └─rtables::export_as_txt(...)
   30. │         └─formatters::paginate_to_mpfs(...)
   31. │           └─formatters::matrix_form(obj, TRUE, TRUE, indent_size = indent_size)
   32. └─base::.handleSimpleError(...)
   33.   └─base (local) h(simpleError(msg, call))
  
  [ FAIL 5 | WARN 0 | SKIP 2 | PASS 179 ]
```

**Comment by Melkiades** [comment]
There is also a difference in the pagination machinery. Now, it seems to drop the top left material as can be seen here:
```r
  clw <- c(10, 8, 8, 10) # random values to have "some" wrapping
  lpp_tmp <- 41 # minimum for this table

  pg_tbl_w_clw <- paginate_table(res, lpp = lpp_tmp, colwidths = clw)
  pg_tbl_no_clw <- paginate_table(res, lpp = lpp_tmp)
  res1 <- toString(pg_tbl_no_clw[[1]], widths = clw[1:3])
  res2 <- toString(res[
    1:17,
    1:2,
    keep_titles = TRUE,
    keep_footers = TRUE,
    keep_topleft = TRUE
  ], widths = clw[1:3])
```

image

blocking https://github.com/insightsengineering/scda.test/issues/42

**Review by Melkiades — CHANGES_REQUESTED** [review]
My apologies for blocking this PR again but there are a couple of (I think small) bugs in `scda.test` https://github.com/insightsengineering/scda.test/pull/44. They are mainly in my finicky tests for tables. They are all highlighted in my PR's comments with (xxx)

**Comment by gmbecker** [comment]
> My apologies for blocking this PR again but there are a couple of (I think small) bugs in `scda.test` [insightsengineering/scda.test#44](https://github.com/insightsengineering/scda.test/pull/44). They are mainly in my finicky tests for tables. They are all highlighted in my PR's comments with (xxx)

I believe these are all either fixed, or the new behavior is correct and the old behavior was wrong. Please confirm @Melkiades

File: NEWS.md
Comment (gmbecker) [reply NEWS.md:8]: It doesn't make sense to me to change one line starting with New but not change the very next entry which starts the same way. I don't really care which way we do it, but if we're going to care about this kind of thing, we should be consistent.

**Review by shajoezhu — APPROVED** [review]
Thanks for the change @gmbecker ! @ayogasekaram  have checked that it is working correctly downstream. and tested in Chevron as well by @clarkliming . Can we please address the remaining comment and unblock the merge. Thanks! @Melkiades anything else?

**Comment by gmbecker** [comment]
> Thanks for the change @gmbecker ! @ayogasekaram have checked that it is working correctly downstream. and tested in Chevron as well by @clarkliming . Can we please address the remaining comment and unblock the merge. Thanks! @Melkiades anything else?

@shajoezhu I don't have the ability to merge this. @Melkiades has to change his review

File: NEWS.md
Comment (gmbecker) [reply NEWS.md:3]: This one is actually wrong and we should not make the change.

**Comment by gmbecker** [comment]
> hi @gmbecker , how about now?

@shajoezhu No. Once a "changes requested" review is returned the only way to unblock without admin status (which I have but don't use in rtables but do not have at all for formatters) is for that person to return a review of approved.

**Comment by shajoezhu** [comment]
> > hi @gmbecker , how about now?
> 
> @shajoezhu No. Once a "changes requested" review is returned the only way to unblock without admin status (which I have but don't use in rtables but do not have at all for formatters) is for that person to return a review of approved.

Got it! Thanks @gmbecker , let me ping @Melkiades a message, and we will have this merged in first thing. Thanks!

**Comment by gmbecker** [comment]
> > hi @gmbecker , how about now?
> 
> @shajoezhu No. Once a "changes requested" review is returned the only way to unblock without admin status (which I have but don't use in rtables but do not have at all for formatters) is for that person to return a review of approved.

It actually looks like I may be able to dismiss his review. Would you like me to do that, @shajoezhu? I dont' know exactly what will happen, because i've never done that before.

File: NEWS.md
Comment (shajoezhu) [reply NEWS.md:3]: hi @gmbecker , can you update the correct wording here. Thanks

Comment (gmbecker) [reply NEWS.md:3]: I do not understand what you mean. the wording is correct before the change. the 'now correctly' is there because they were documented as inches before but interpreted as lines in at least some places.

Comment (shajoezhu) [reply NEWS.md:3]: I see. Thanks @gmbecker , ok, we can mark this one as resolved then.

**Review by Melkiades — APPROVED** [review]
Great work Gabe! Thanks :) gonna merge it now

**Pull request merged**
